### ISSUE TITLE
Apache proxy ssl dont work.

### ISSUE THREAD

**Issue by ernestorumo** [butlerx/wetty#93]
**Apache proxy:**
```
ProxyPreserveHost On
        AllowEncodedSlashes On
        ProxyPass /wetty/ http://ernestorumo.es:9109/
        ProxyPassReverse /wetty/ http://ernestorumo.es:9109/
        ProxyHTMLURLMap http://ernestorumo.es:9109/ /wetty/
        <Location /wetty/>
                ProxyPassReverse /
                ProxyHTMLEnable On
                ProxyHTMLURLMap / /wetty/
        </Location>

```
The page don´t load and the error it´s the following:

Request URL:https://ernestorumo.es/wetty/socket.io/?EIO=3&transport=polling&t=Ls62JZc
Request Method:GET
Status Code:404 Not Found
Remote Address:176.31.121.174:443
Referrer Policy:no-referrer-when-downgrade

**Comment by mirtouf** [comment]
Hello,

this seems to work (most important part):
```
        RewriteCond %{REQUEST_URI}  ^/wetty/socket.io [NC]
        RewriteCond %{QUERY_STRING} transport=websocket [NC]
        RewriteRule /wetty/socket.io/(.*) ws://localhost:9123/wetty/socket.io/$1 [P,L]

        <LocationMatch ^/wetty/(.*)>
                DirectorySlash On
                Require all granted
                ProxyPassMatch http://127.0.0.1:9123
                ProxyPassReverse /wetty/
                #LimitRequestBody 1048576000
        </LocationMatch>
```
I access on https://mydomain.tld/wetty/
Just modify the port for your purpose. I did this today, so it might change later depending on the results of my tests.
Maybe filtering on the headers is nicer:
https://stackoverflow.com/questions/27526281/websockets-and-apache-proxy-how-to-configure-mod-proxy-wstunnel

**Comment by butlerx** [comment]
@mirtouf is now the recommended way to use apache and in the readme

**Comment by mirtouf** [comment]
Due to recent changes, this setup is no more working, it has to be modified.

**Comment by butlerx** [comment]
@mirtouf mind explaining what needs to be changed or updating the readme

**Comment by mirtouf** [comment]
I am working on it when I will have more time. At the moment, it fails with infinite loop redirections as wetty is no more accessible through http://localhost:3000/ but http://localhost:3000/wetty/

**Comment by butlerx** [comment]
I imagine this is caused by a combination of a few changes. The server does a 301 redirect to for trailing slashes. Aswell as not having duplicate routes so wetty is only served on `/wetty` this is controlled thought the base `--base flag`

**Comment by mirtouf** [comment]
Yep, this is the behaviour I noticed.

**Comment by mirtouf** [comment]
Is it possible to prevent wetty to do a redirection with a dedicated flag or to add such a function ?

**Comment by butlerx** [comment]
Yeah we can add a flag for users to disable the middleware if you are going to run it be ing a proxy

**Comment by andrew-gw** [comment]
This doesn't work. I followed all instructions, and Apache fails to load. It just says "failed!". I put it in `/etc/apache2/apache2.conf`. If I try to use Wetty without this config, I see nothing but the directory contents, even with the port number like `localhost:3000/wetty`. This whole package just seems not to work whatsoever. I git cloned, I npm installed, I yarned, I followed all the instructions. This Apache configuration is the only thing missing I suppose, but the supplied one just doesn't work.
